[nettle] fails to build an x64-osx triplet on Apple Silicon

**Host Environment**
- OS: macOS 11.2.2 (20D80) 
- Compiler: Apple clang version 12.0.0 (clang-1200.0.32.29)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install nettle`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /bin/bash ./../src/48e146e54f-e59f545639.clean/configure --build=arm64-apple-darwin --host=x86_64-apple-darwin --disable-documentation --disable-openssl --disable-shared --disable-silent-rules --verbose --disable-shared --enable-static --prefix=/Users/cromo/Documents/Sourcecode/vcpkg/installed/x64-osx/debug --bindir=${prefix}/../tools/nettle/debug/bin --sbindir=${prefix}/../tools/nettle/debug/sbin --includedir=${prefix}/../include --datarootdir=${prefix}/share/nettle
```
`Sourcecode/vcpkg/buildtrees/nettle/x64-osx-dbg/config-x64-osx-dbg-err.log`:
```
Invalid configuration `arm64-apple-darwin': machine `arm64-apple' not recognized
configure: error: /bin/sh ./../src/48e146e54f-e59f545639.clean/./config.sub arm64-apple-darwin failed
```

**Additional context**
Notice that this is a x64 (Rosetta) build, but the `arm64-apple-darwin` triplet is being incorrectly passed to `configure`.

@PhoebeHui just did, fails with the same error:

```
Invalid configuration `arm64-apple-darwin': machine `arm64-apple' not recognized
configure: error: /bin/sh ./../src/48e146e54f-e59f545639.clean/./config.sub arm64-apple-darwin failed
```

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -438,8 +438,9 @@ function(vcpkg_configure_make)
             # https://stackoverflow.com/questions/21990021/how-to-determine-host-value-for-configure-when-using-cross-compiler
             # Only for ports using autotools so we can assume that they follow the common conventions for build/target/host
             if(NOT "${TARGET_ARCH}" STREQUAL "${BUILD_ARCH}") # we don't need to specify the additional flags if we build natively.
-                list(APPEND "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
+                set(_csc_BUILD_TRIPLET "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
             endif()
+            debug_message("Using make triplet: ${_csc_BUILD_TRIPLET}")
         endif()
     endif()
 

--- a/scripts/cmake/vcpkg_configure_make.cmake
+++ b/scripts/cmake/vcpkg_configure_make.cmake
@@ -437,12 +437,9 @@ function(vcpkg_configure_make)
             # --target: the machine that CC will produce binaries for
             # https://stackoverflow.com/questions/21990021/how-to-determine-host-value-for-configure-when-using-cross-compiler
             # Only for ports using autotools so we can assume that they follow the common conventions for build/target/host
-            set(_csc_BUILD_TRIPLET "--build=${BUILD_ARCH}-apple-darwin")
             if(NOT "${TARGET_ARCH}" STREQUAL "${BUILD_ARCH}") # we don't need to specify the additional flags if we build natively.
-
-                list(APPEND _csc_BUILD_TRIPLET "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
+                list(APPEND "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
             endif()
-            debug_message("Using make triplet: ${_csc_BUILD_TRIPLET}")
         endif()
     endif()
 

